sdk/appconfiguration/azure-data-appconfiguration/inc/azure/data/appconfiguration/configuration_client_options.hpp

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+#pragma once
+
+#include "configuration_client_models.hpp"
+#include "dll_import_export.hpp"
+
+#include <azure/core/internal/client_options.hpp>
+#include <azure/core/internal/extendable_enumeration.hpp>
+#include <azure/core/nullable.hpp>
+#include <azure/core/url.hpp>
+#include <azure/core/uuid.hpp>
+
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace Azure { namespace Data { namespace AppConfiguration {
+  class KeyValueFields;
+  struct KeyValue;
+  class SnapshotFields;
+  class SnapshotStatus;
+  class LabelFields;
+
+  struct ConfigurationClientOptions final : public Azure::Core::_internal::ClientOptions
+  {
+    std::string ApiVersion = "2023-11-01";
+  };
+
+  struct GetKeysOptions final
+  {
+    std::string NextPageToken;
```

Comment:
You are right.
These are all marked as optional in the tsp, which is likely why they are in the `Options` bag rather than as method parameters.

I suspect this is an emitter issue where `string?` isn't being emitted as `Nullable<std::string>`:
https://github.com/Azure/azure-rest-api-specs/blob/7dfd17073198d5cf3df99a106b770cb01b395f2a/specification/appconfiguration/AppConfiguration/routes.tsp#L50-L82